### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by surazad99**
Category CRUD completed AND Club Creation APi done 

Part of #31

### CODE REVIEW

File: backend/app/Http/Controllers/Api/CategoryController.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+     */
+    public function show($id)
+    {
+        try {
+            $category = Category::find($id);
+            if (!$category)
+                throw new Exception('Data Not Found');
+
+            $data['message'] = 'Success';
+            $data['status'] = 201;
```
Comment (kiranparajuli589) [inline backend/app/Http/Controllers/Api/CategoryController.php:70]:
```suggestion
            $data['status'] = 200;
```

File: backend/app/Http/Controllers/Api/CategoryController.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(CategoryRequest $request, Category $category)
+    {
+        try {
+            $validated = $request->validated();
+            $category->update($validated);
+            return response()->json(['message' => 'Category Updated Successfully', 'status' => 201]);
```
Comment (kiranparajuli589) [inline backend/app/Http/Controllers/Api/CategoryController.php:93]:
```suggestion
            return response()->json(['message' => 'Category Updated Successfully', 'status' => 204]);
```

File: backend/app/Http/Controllers/Api/CategoryController.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy(Category $category)
+    {
+        try {
+            $category->delete();
+            return response()->json(['message' => 'Category Deleted Successfully', 'status' => 201]);
```
Comment (kiranparajuli589) [inline backend/app/Http/Controllers/Api/CategoryController.php:113]:
```suggestion
            return response()->json(['message' => 'Category Deleted Successfully', 'status' => 204]);
```

File: backend/app/Http/Controllers/Api/DistrictController.php [hunk 4]
```diff
@@ -65,21 +60,19 @@ public function store(DistrictRequest $request)
      */
     public function show($id)
     {
-        try{
+        try {
             $district = District::find($id);
-            if(!$district)
+            if (!$district)
                 throw new Exception('Data Not Found');
```
Comment (kiranparajuli589) [inline backend/app/Http/Controllers/Api/DistrictController.php:66]: throw HTTP response with a similar message
frontend ma error handle garna easy hunxa

File: backend/app/Http/Controllers/Api/DistrictController.php [hunk 5]
```diff
@@ -92,19 +85,17 @@ public function show($id)
...
     public function update(DistrictRequest $request, District $district)
     {
-        try{
+        try {
             $validated = $request->validated();
             $district->update($validated);
-            return response()->json(['message' => 'District Created Successfully', 'status' => 201]);
-
-        }catch(Exception $e){
+            return response()->json(['message' => 'District Updated Successfully', 'status' => 201]);
```
Comment (kiranparajuli589) [inline backend/app/Http/Controllers/Api/DistrictController.php:91]: use `204` for update
Ani yo `put` ho ki `patch` ho?

File: backend/app/Http/Controllers/Api/DistrictController.php [hunk 6]
```diff
@@ -115,17 +106,14 @@ public function update(DistrictRequest $request, District $district)
      */
     public function destroy(District $district)
     {
-        try{
+        try {
             $district->delete();
-            return response()->json(['message' => 'District Deleted Successfully','status' => 201]);
-
-        }catch(Exception $e){
+            return response()->json(['message' => 'District Deleted Successfully', 'status' => 201]);
```
Comment (kiranparajuli589) [inline backend/app/Http/Controllers/Api/DistrictController.php:111]: use `200` for delete success

File: backend/app/Http/Requests/CategoryRequest.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+class CategoryRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
```
Comment (kiranparajuli589) [inline backend/app/Http/Requests/CategoryRequest.php:16]: only returning true :confused: :thinking: ??

File: backend/app/Http/Requests/ClubRequest.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
```
Comment (kiranparajuli589) [inline backend/app/Http/Requests/ClubRequest.php:17]: same function in multiple files... 
DRY yourself Mr....IMO we call this authorize form some helper.php

Comment (surazad99) [reply backend/app/Http/Requests/ClubRequest.php:17]: Gadha laravel framework le dine Class ho yo ......sab ma huncha authorize ma true rakhena vane ta sabai api forbidden huncha

File: backend/app/Http/Controllers/Api/CategoryController.php (hunk 3 above)
Comment (surazad99) [reply backend/app/Http/Controllers/Api/CategoryController.php:113]: Default Success ma 200 aaucha tyo server le pathaune ho ......ani custom code pathako every success ma 201 vanera ...sabai api ma success vayepaxi aautai code aauxa ...delete update ma xuta xutai code pathayo vane confuse huncha frontend ma use garna lai

Comment (kiranparajuli589) [reply backend/app/Http/Controllers/Api/CategoryController.php:113]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
hawa j rkhde pani hunna

**Pull request merged**